roles/create_vms/templates/create_vm.sh.j2

```diff
@@ -20,8 +25,12 @@ virt-install \
     --noautoconsole \
     --wait=-1 \
     {% if not is_on_rhel9 %}
+    {% if create_vms_disable_secure_boot %}
+    --boot loader=/usr/share/edk2/ovmf/OVMF_CODE.secboot.fd,loader.readonly=yes,loader.secure='no',loader.type=pflash,nvram=/var/lib/libvirt/qemu/nvram/{{ item.name }}_no_secure_boot_VARS.fd \
```

Comment:
>   \"message\": \"Unknown boot loader path \\\"/usr/share/edk2/ovmf/OVMF_CODE.secboot.fd\\\" in domain \\\"fc749263-ef08-421c-9833-c01c20f3cf4f\\\" configuration encountered while setting boot mode \\\"UEFI\\\", system architecture \\\"x86_64\\\". Consider adding this loader path to emulator config.\",

I see the Hypervisors default loader points to `/usr/share/OVMF/OVMF_CODE.secboot.fd`

```
[dci@server09 ~]$ sudo virsh domcapabilities              
<domainCapabilities>                           
  <path>/usr/libexec/qemu-kvm</path>                  
  <domain>kvm</domain>                                          
  <machine>pc-i440fx-rhel7.6.0</machine>            
  <arch>x86_64</arch>                             
  <vcpu max='240'/>                              
  <iothreads supported='yes'/>                   
  <os supported='yes'>                           
    <enum name='firmware'/>                                
    <loader supported='yes'>                   
      <value>/usr/share/OVMF/OVMF_CODE.secboot.fd</value>
      <enum name='type'>                         
        <value>rom</value>                  
        <value>pflash</value>               
      </enum>                                             
      <enum name='readonly'>                       
        <value>yes</value>                            
        <value>no</value>                        
      </enum>                                  
      <enum name='secure'>                
        <value>no</value>              
      </enum>                             
    </loader>                             
  </os>  
...  
```
And that path is a symbolic link to `/usr/share/edk2/ovmf/OVMF_CODE.secboot.fd` 
```
[dci@server09 ~]$ ls -l /usr/share/OVMF/OVMF_CODE.secboot.fd
lrwxrwxrwx. 1 root root 33 Mar 14 08:56 /usr/share/OVMF/OVMF_CODE.secboot.fd -> ../edk2/ovmf/OVMF_CODE.secboot.fd
```
So it is the same file, but in your xml config the loader points to `/usr/share/edk2/ovmf/OVMF_CODE.secboot.fd` so I wonder if you change it to match the same path as shown in the `virsh domcapabilities` output, that is to use the path `/usr/share/OVMF/OVMF_CODE.secboot.fd` everywhere to be consistent. 

I manually powered a VM from the last job failure, and it booted, so I got the impression all looks good in libvirt, it's only sushy that complains right?, where did you get that error @ramperher ? 

Initially I had some concerns that OVMF_CODE.secboot.fd only works with secure boot enabled, as Andreas described it in this blog https://andreaskaris.github.io/blog/linux/libvirt-uefi-without-secureboot/ where he even uses an non-secure boot UEFI firmware, but it might be an old bug.

Reply:
I see now where did you get the error, when KVM is trying to Set OneTimeBoot Virtual Media. I went to the logs carefully, so it's really sushy that complains : )

I think I managed to pass that error, long story short, I tested my theory above and if you change `--boot loader=/usr/share/edk2/ovmf/OVMF_CODE.secboot.fd` to `--boot loader=/usr/share/OVMF/OVMF_CODE.secboot.fd` in the VM definition, then we are able to set the OneTimeBoot.

I replicated the issue with a basic play, using dciokd-master-1 in server09:
```
---
- name: Play to validate stuff
  hosts: localhost
  gather_facts: false
  vars:
    system_uuid_url: "https://server09-assisted.partnerci.bos2.lab:8082/redfish/v1/Systems/fc749263-ef08-421c-9833-c01c20f3cf4f"                                                                                                             
    bmc_user: USER
    bmc_password: PASSWORD
    target_host: dciokd-master-1
  tasks:

    - name: KVM Set OneTimeBoot Virtual Media {{ target_host }}
      uri:
        url: "{{ system_uuid_url }}"
        user: "{{ bmc_user }}"
        password: "{{ bmc_password }}"
        method: PATCH
        body_format: json
        body:
          {
            "Boot":
              {
                "BootSourceOverrideTarget": "Cd",
                "BootSourceOverrideMode": "UEFI",
                "BootSourceOverrideEnabled": "Continuous",
              },
          }
        status_code: [200, 204]
        force_basic_auth: false
        validate_certs: false
        return_content: true

[dci@server09 ~]$ sudo virsh dumpxml dciokd-master-1 | grep uuid
  <uuid>fc749263-ef08-421c-9833-c01c20f3cf4f</uuid>

```

Then I tried to set the OneTimeBoot when the VM had a loader pointing to `/usr/share/edk2/ovmf/OVMF_CODE.secboot.fd`  and it fails.
```
[dci@server09 ~]$ sudo virsh dumpxml dciokd-master-1 | grep loader
    <loader readonly='yes' secure='no' type='pflash'>/usr/share/edk2/ovmf/OVMF_CODE.secboot.fd</loader>

[dci-openshift-agent@server01 manny]$ ansible-playbook simple-play.yml                                                                                                                                                                       
PLAY [Play to validate stuff] ****************************************************************************************************************************************************************************************************************

TASK [KVM Set OneTimeBoot Virtual Media dciokd-master-1] *************************************************************************************************************************************************************************************
fatal: [localhost]: FAILED! => {"changed": false, "content": "{\n  \"error\": {\n    \"code\": \"Base.1.0.GeneralError\",\n    \"message\": \"Unknown boot loader path \\\"/usr/share/edk2/ovmf/OVMF_CODE.secboot.fd\\\" in domain \\\"fc749263-ef08-421c-9833-c01c20f3cf4f\\\" configuration encountered while setting boot mode \\\"UEFI\\\", system architecture \\\"x86_64\\\". Consider adding this loader path to emulator config.\",\n    \"@Message.ExtendedInfo\": [\n      {\n        \"@odata.type\": \"/redfish/v1/$metadata#Message.1.0.0.Message\",\n        \"MessageId\": \"Base.1.0.GeneralError\"\n      }\n    ]\n  }\n}", "content_length": "510", "content_type": "application/json", "date": "Thu, 04 Jul 2024 23:30:54 GMT", "elapsed": 0, "json": {"error": {"@Message.ExtendedInfo": [{"@odata.type": "/redfish/v1/$metadata#Message.1.0.0.Message", "MessageId": "Base.1.0.GeneralError"}], "code": "Base.1.0.GeneralError", "message": "Unknown boot loader path \"/usr/share/edk2/ovmf/OVMF_CODE.secboot.fd\" in domain \"fc749263-ef08-421c-9833-c01c20f3cf4f\" configuration encountered while setting boot mode \"UEFI\", system architecture \"x86_64\". Consider adding this loader path to emulator config."}}, "msg": "Status code was 500 and not [200, 204]: HTTP Error 500: INTERNAL SERVER ERROR", "redirected": false, "server": "Werkzeug/2.0.3 Python/3.6.8", "status": 500, "url": "https://server09-assisted.partnerci.bos2.lab:8082/redfish/v1/Systems/fc749263-ef08-421c-9833-c01c20f3cf4f"}

PLAY RECAP ***********************************************************************************************************************************************************************************************************************************
localhost                  : ok=0    changed=0    unreachable=0    failed=1    skipped=0    rescued=0    ignored=0
```

Then, I changed the loader to point to /usr/share/OVMF/OVMF_CODE.secboot.fd, and this time the play passed
```
[dci@server09 ~]$ sudo virsh edit dciokd-master-1
Domain 'dciokd-master-1' XML configuration edited.

[dci@server09 ~]$ sudo virsh dumpxml dciokd-master-1 | grep loader
    <loader readonly='yes' secure='no' type='pflash'>/usr/share/OVMF/OVMF_CODE.secboot.fd</loader>

[dci-openshift-agent@server01 manny]$ ansible-playbook simple-play.yml                                                                                                                                                                       
PLAY [Play to validate stuff] ****************************************************************************************************************************************************************************************************************

TASK [KVM Set OneTimeBoot Virtual Media dciokd-master-1] *************************************************************************************************************************************************************************************
ok: [localhost]

PLAY RECAP ***********************************************************************************************************************************************************************************************************************************
localhost                  : ok=1    changed=0    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0
```

Then this makes me believe that in the VM definition we need to use the same loader path as defined in sushy config:
```
[dci@server09 ~]$ grep -A4 SUSHY_EMULATOR_BOOT_LOADER_MAP /opt/dci/assisted/sushy/sushy-emulator.conf 
SUSHY_EMULATOR_BOOT_LOADER_MAP = {
    u'UEFI': {
        u'x86_64': u'/usr/share/OVMF/OVMF_CODE.secboot.fd',
        u'aarch64': u'/usr/share/AAVMF/AAVMF_CODE.fd'
    },
```

Reply:
Good observation @manurodriguez , thanks for checking this :) I'll change the boot loader to use `/usr/share/OVMF/OVMF_CODE.secboot.fd` now and let's see.

Reply:
Now KVM failed task is now passing, I think this can be closed, let's see if the job passes.